components-compose/src/main/java/uk/gov/hmrc/components/compose/molecule/input/PasswordInputView.kt

```diff
@@ -0,0 +1,376 @@
+    val indicatorColor = indicatorColor(enabled, isError, interactionSource)
+    val targetThickness = if (focused) focusedBorderThickness else unfocusedBorderThickness
+    val animatedThickness = if (enabled) {
+        animateDpAsState(targetThickness, tween(durationMillis = ANIMATION_DURATION))
+    } else {
+        rememberUpdatedState(unfocusedBorderThickness)
+    }
+    return rememberUpdatedState(
+        BorderStroke(animatedThickness.value, SolidColor(indicatorColor.value))
+    )
+}
+
+var borderStroke: State<BorderStroke>? = null
+
+@Composable
+fun CalculateBorder(
+    enabled: Boolean = true,
+    isError: Boolean,
+    interactionSource: InteractionSource,
+    focusedBorderThickness: Dp = FocusedBorderThickness,
+    unfocusedBorderThickness: Dp = UnfocusedBorderThickness
+) {
+    borderStroke = animateBorderStrokeAsState(
+        enabled,
+        isError,
+        interactionSource,
+        focusedBorderThickness,
+        unfocusedBorderThickness
+    )
+}
+
+@Composable
+fun indicatorColor(
+    enabled: Boolean,
+    isError: Boolean,
+    interactionSource: InteractionSource
+): State<Color> {
+    val focused by interactionSource.collectIsFocusedAsState()
+
+    val colors = HmrcTheme.colors
+
+    val focusedBorderColor = colors.hmrcBlue
+    val errorBorderColor = colors.hmrcBlack
+    val unfocusedBorderColor = colors.hmrcBlack
+
+    val targetValue = when {
+        isError -> errorBorderColor
+        focused -> focusedBorderColor
+        else -> unfocusedBorderColor
+    }
+    return if (enabled) {
+        animateColorAsState(targetValue, tween(durationMillis = ANIMATION_DURATION))
+    } else {
+        rememberUpdatedState(targetValue)
+    }
+}
+
+@SuppressWarnings("LongMethod", "ComplexMethod")
+@Composable
+fun PasswordTextInputView(
+    modifier: Modifier = Modifier,
+    value: String? = null,
+    onInputValueChange: ((String) -> Unit)? = null,
+    inputFilter: ((String, String) -> String)? = null,
+    labelText: String? = null,
+    labelContentDescription: String? = null,
+    hintText: String? = null,
+    hintContentDescription: String? = null,
+    prefix: @Composable() (() -> Unit)? = null,
+    placeholderText: String? = null,
+    errorText: String? = null,
+    errorContentDescription: String? = null,
+    showButtonText: String? = stringResource(R.string.passwordTextInputView_show),
+    hideButtonText: String? = stringResource(R.string.passwordTextInputView_hide),
+    showButtonContentDescription: String? = null,
+    hideButtonContentDescription: String? = null,
+    showButtonStateDescription: String? = null,
+    hideButtonStateDescription: String? = null,
+    characterCount: Int? = null,
+    maxChars: Int? = null,
+    singleLine: Boolean = false,
+    keyboardOptions: KeyboardOptions = KeyboardOptions.Default,
+    interactionSource: MutableInteractionSource,
+    requiredSequencesSpacing: Boolean,
+    textStyle: TextStyle
```

Comment:
Same comments as above for these parameters.. do we need them all? Some of them we can hardcode as we don't want the user to be able to change them. Although this function isn't exposed to the user, we can save code by defining our set rules at the earliest point, which is inside the `TextField` declaration in this function.